Fixing comment plugin not using user settings when overriding default setting (#3424)
zyedidia/micro: runtime/plugins/comment/help/comment.md

* zsh: `# %s`

If your filetype is not available here, you can simply modify
the `comment.type` option:

```
set comment.type "/* %s */"
```

Or in your `settings.json`:

```json
{
    "*.c": {
        "comment.type": "/* %s */"
    }
}
```

`commenttype` (without the dot) is the legacy option that is superseded by `comment.type`.
`commenttype` is still supported but will get deprecated in the future.

> **Use `comment.type` instead.**

Reviewer: It's better to stick to the rule of wrapping lines to not exceed a certain number of characters per line. Generally in micro's help files this limit is around 80 characters. In this particular file, as I see, it is even less (more like 65 characters), so it's better to stay consistent with that.

* zsh: `# %s`

If your filetype is not available here, you can simply modify
the `comment.type` option:

```
set comment.type "/* %s */"
```

Or in your `settings.json`:

```json
{
    "*.c": {
        "comment.type": "/* %s */"
    }
}
```

> `commenttype` was the previous option name that was replaced by `comment.type`.

Reviewer: Sounds a bit unclear and torn out of some context.

How about something like: "For backward compatibility, use can also use the option name `commenttype` (without the dot) instead of `comment.type`. It is recommended to use `comment.type` instead, as `commenttype` can get deprecated in the future."

Author: This is pretty minor but how about 

``` text
`commenttype` (without the dot) is the legacy option that is superseded by `comment.type`.

`commenttype` is still supported but will get deprecated in the future. 

**Use `comment.type` instead.**
```

Just firmer wording so that we can discourage people from using `commenttype`.

Reviewer 2: Shouldn't it just say `commenttype` is deprecated? "Will get deprecated in the future" doesn't make much sense to me.

> Features are deprecated, rather than immediately removed, to provide backward compatibility and to give programmers time to bring affected code into compliance with the new standard.

Reviewer 3: > Shouldn't it just say `commenttype` is deprecated?

I agree.
It is deprecated in the moment of merge.

superseded by `comment.type`. `commenttype` is still supported

> but deprecated from now on.

Reviewer: What does "now" mean to someone who is ready this, say, 3 years from now?

Author: Updated.

> superseded by `comment.type`. `commenttype` is still supported
> but will get deprecated in the future.

Reviewer 3: @Neko-Box-Coder:
As mentioned [here](https://github.com/zyedidia/micro/pull/3424#discussion_r2106326952), can you adapt it to something like:
```suggestion
superseded by `comment.type`. `commenttype` is still supported
but deprecated from now on.
```
